quarto/src/Repository/GameRepository.php

```diff
@@ -39,22 +44,43 @@ public function getOpenedGamesList(array $tokenList)
 
     public function getCurrentGamesList(array $tokenList)
     {
-        $query = $this->em->createQuery("SELECT g FROM App:Game g  " .
+        $queryOne = $this->em->createQuery("SELECT g.id_game idGame,
+            g.number_players numberPlayers,
+            g.solo_game soloGame,
+            g.closed,
+            g.token_player_one token
+        FROM App:Game g  " .
         "WHERE g.closed=false  " .
-        "AND (g.token_player_one IN (:player_one_token)  " .
-        "  OR g.token_player_two IN (:player_two_token))  " .
+        "AND g.token_player_one IN (:player_one_token) " .
         "ORDER BY g.id_game ASC");
-        $query->setParameter('player_one_token', $tokenList);
-        $query->setParameter('player_two_token', $tokenList);
-        return $query->getResult();
+        $queryOne->setParameter('player_one_token', $tokenList);
+        $arrayOne = $queryOne->getResult();
+
+        $queryTwo = $this->em->createQuery("SELECT g.id_game idGame,
+            g.number_players numberPlayers,
+            g.solo_game soloGame,
+            g.closed,
+            g.token_player_two token
+        FROM App:Game g  " .
+        "WHERE g.closed=false  " .
+        "AND g.token_player_two IN (:player_two_token) " .
+        "ORDER BY g.id_game ASC");
+        $queryTwo->setParameter('player_two_token', $tokenList);
+        $arrayTwo = $queryTwo->getResult();
+        return array_merge($arrayOne, $arrayTwo);
     }
 
     public function getOnlySpectateGamesList(array $tokenList)
     {
         if (count($tokenList) === 0) {
             $tokenList=["NOTHING"];
         }
-        $query = $this->em->createQuery("SELECT g FROM App:Game g  " .
+        $query = $this->em->createQuery("SELECT g.id_game idGame,
+            g.number_players numberPlayers,
+            g.solo_game soloGame,
+            g.closed,
+            '' token
```

Comment:
??

Reply:
Returns structure have to be identical in all lists requests, so they can be managed the same way by clients.
For watch only games list, the user have no token (because he is not registerd in the game)
So I return an empty field for token